src/node/executor.cpp

```diff
@@ -86,66 +93,96 @@ void Executor::execute_(PbftBlock const &pbft_block) {
+
     for (auto const &dag_blk_raw : dag_blks_raw) {
-      for (auto const &trx_h : DagBlock::extract_transactions_from_rlp(RLP(dag_blk_raw))) {
-        if (!unique_trxs.insert(trx_h).second) {
+      const auto &dag_blk_rlp = RLP(dag_blk_raw);
+      DagBlock dag_block(dag_blk_rlp);  // Full block has to be decoded from rlp to get block author from signature
+      // TODO: Check if dag block is validated to consist at least 1 tx. Is this checked before ???
+
+      addr_t dag_block_author = dag_block.getSender();
+
+      for (auto const &tx_hash : dag_block.getTrxs()) {
+        if (dag_stats_tmp_buf_.addTransaction(tx_hash, dag_block_author)) {
+          // Tx was already processed
           continue;
         }
-        db_query.append(DbStorage::Columns::executed_transactions, trx_h);
-        db_query.append(DbStorage::Columns::transactions, trx_h);
+
+        db_query.append(DbStorage::Columns::executed_transactions, tx_hash);
+        db_query.append(DbStorage::Columns::transactions, tx_hash);
       }
+
+      // TODO: there is a possibility that some miner includes only txs that are later in filtered out due to
+      // replay_protection_service_ or "has_been_executed", but gets rewards for dag block creation
+      // It might or might not be ok, he provided valid block and validated therefore other blocks so he did
+      // some work, on the other hand none of the txs that he included were actually processed...
+      dag_stats_tmp_buf_.addDagBlock(dag_block_author);
     }
+
     auto trx_db_results = db_query.execute(false);
-    for (uint i = 0; i < unique_trxs.size(); ++i) {
-      auto has_been_executed = !trx_db_results[0 + i * 2].empty();
+    for (uint i = 0; i < trx_db_results.size() / 2; ++i) {
+      size_t executed_tx_result_idx = 0 + i * 2;
+      size_t tx_result_idx = 1 + i * 2;
+
+      h256 tx_hash = h256(db_query.get_key(tx_result_idx));
+
+      auto has_not_been_saved = trx_db_results[tx_result_idx].empty();
+      if (has_not_been_saved) {
+        // This should never happen
+        throw std::runtime_error("Executor::executePbftBlocks_: invalid transaction is being processed: tx_hash: " +
+                                 tx_hash.toString());
+      }
+
+      bool has_been_executed = !trx_db_results[executed_tx_result_idx].empty();
       if (has_been_executed) {
         continue;
       }
-      // Non-executed trxs
-      auto const &trx = transactions_tmp_buf_.emplace_back(&trx_db_results[1 + i * 2], dev::eth::CheckTransaction::None,
-                                                           true, h256(db_query.get_key(1 + i * 2)));
-      if (replay_protection_service_->is_nonce_stale(trx.sender(), trx.nonce())) {
-        transactions_tmp_buf_.pop_back();
+
+      // Non-executed txs
+      auto tx = dev::eth::Transaction(&trx_db_results[tx_result_idx], dev::eth::CheckTransaction::None, true, tx_hash);
+
+      if (replay_protection_service_->is_nonce_stale(tx.sender(), tx.nonce())) {
```

Comment:
we conventionally call transactions "trx" - haven't you noticed?

Reply:
you are right, will fix that